lib/mongoid/threaded.rb

```diff
@@ -197,27 +197,37 @@ def client_override=(name)
     # Get the current Mongoid scope.
     #
     # @example Get the scope.
-    #   Threaded.current_scope
+    #   Threaded.current_scope(klass)
     #
     # @return [ Criteria ] The scope.
     #
     # @since 5.0.0
-    def current_scope
-      Thread.current[CURRENT_SCOPE_KEY]
+    def current_scope(klass)
+      if Thread.current[CURRENT_SCOPE_KEY]
+        klass_key = klass
+        Thread.current[CURRENT_SCOPE_KEY].each_key { |key| klass_key = key if key <= klass }
+        Thread.current[CURRENT_SCOPE_KEY][klass_key]
+      end
```

Comment:
Another, more Ruby-like way of doing this logic would be using the #find method. http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-find

You could do:

```ruby
Thread.current[CURRENT_SCOPE_KEY].find do |k, v|
  k <= klass
end
```

This will test each key and value pair and return the first pair for which k <= klass returns true.